service/src/main/java/com/commercetools/pspadapter/payone/IntegrationService.java

```diff
@@ -84,16 +84,33 @@ private static void initTenantServiceResources(final TenantFactory tenantFactory
         if (StringUtils.isNotEmpty(paymentHandlerUrl)) {
             LOG.info("Register payment handler URL {}", paymentHandlerUrl);
             Spark.get(paymentHandlerUrl, (req, res) -> {
-                    final PaymentHandleResult paymentHandleResult = paymentHandler.handlePayment(req.params("id"));
-                    if (!paymentHandleResult.body().isEmpty()) {
+                    final PayoneResult payoneResult = paymentHandler.handlePayment(req.params("id"));
+                    if (!payoneResult.body().isEmpty()) {
                         LOG.debug("--> Result body of ${getTenantName()}/commercetools/handle/payments/{}: {}",
-                            req.params("id"), paymentHandleResult.body());
+                            req.params("id"), payoneResult.body());
                     }
-                    res.status(paymentHandleResult.statusCode());
+                    res.status(payoneResult.statusCode());
                     return res;
                 },
                 new HandlePaymentResponseTransformer());
         }
+        // register start Session URL
+        String startSessionUrl = tenantFactory.getPayoneStartSessionUrl();
+        if (StringUtils.isNotEmpty(startSessionUrl)) {
+            LOG.info("Register start session URL {}", startSessionUrl);
+            SessionHandler sessionHandler= tenantFactory.getSessionHandler();
```

Comment:
StartSession is Klarna specific, also "SessionHandler" might be miss-leading.

Reply:
but  maybe we will add other payments to the adapter, which will also need to start a session. Because of that I have choosen a not klarna specific name,.

Reply:
I'd rather go with more meaningful naming for now to make it easy to understand to every DEV. A name is changed quickly when this case will happen sometime.

Reply:
Also, find a better name for "SessionHandler" as it is really misleading

Reply:
8b0331374